pre_built_workers/prepare_preBuild_workers/prepare_for_prebuilt_workers.go

```diff
@@ -0,0 +1,232 @@
+		data.languages["php"] = data.phpGitRef
+	}
+	if _, ok := data.languages["node"]; ok {
+		data.languages["node"] = data.nodeGitRef
+	}
+	if _, ok := data.languages["ruby"]; ok {
+		data.languages["ruby"] = data.rubyGitRef
+	}
+
+	log.Print("Selected language : gitref")
+	formattedMap, _ := json.MarshalIndent(data.languages, "", "  ")
+	log.Print(string(formattedMap))
+
+	for lang, gitRef := range data.languages {
+		var containerRegistry = fmt.Sprintf("%s%s:%s", data.preBuiltImagePrefix, lang, testTag)
+
+		// Build images
+		var dockerfileLocation = fmt.Sprintf("../../containers/pre_built_workers/%s/", lang)
+
+		var buildDockerImage *exec.Cmd
+
+		buildDockerImage = exec.Command(
+			"docker",
+			"build", dockerfileLocation, "-t", containerRegistry, "-build-arg", fmt.Sprintf("GITREF=%s", gitRef),
+		)
+
+		log.Println(fmt.Sprintf("building %s imags", lang))
+
+		buildStdout, err := buildDockerImage.StdoutPipe()
+		if err != nil {
+			log.Fatal(fmt.Sprintf("fail to present logs when building %s image", lang))
+		}
+
+		if err := buildDockerImage.Start(); err != nil {
+			log.Fatal(fmt.Sprintf("failed to build %s worker: %s", lang, err))
+		}
+
+		buildScanner := bufio.NewScanner(buildStdout)
+		buildScanner.Split(bufio.ScanLines)
+		for buildScanner.Scan() {
+			i := buildScanner.Text()
+			fmt.Println(i)
+		}
+		if err := buildDockerImage.Wait(); err != nil {
+			log.Fatal(fmt.Sprintf("exit during buildinig %s image, error: %s", lang, err))
+		}
+
+		log.Println(fmt.Sprintf("successfully build %s worker: %s", lang, containerRegistry))
+
+		// Record current image's registry to the file
+		message := fmt.Sprintf("%s: %s\n", lang, containerRegistry)
+
+		if _, err := writer.WriteString(message); err != nil {
+			log.Fatal(fmt.Sprintf("failed to record %s image with its container registry: %s", lang, containerRegistry))
+		}
+
+		if err := writer.Flush(); err != nil {
+			log.Fatal(fmt.Sprintf("failed to write message (%s) from buffer)", message))
+		}
+
+		// Push image
+		pushDockerImage := exec.Command(
+			"docker",
+			"push",
+			containerRegistry,
+		)
+
+		log.Println(fmt.Sprintf("pushing %s image", lang))
+
+		pushStdout, err := pushDockerImage.StdoutPipe()
+		if err != nil {
+			log.Fatal(fmt.Sprintf("failed to present logs when pushing %s images", lang))
+		}
+
+		if err := pushDockerImage.Start(); err != nil {
+			log.Fatal(fmt.Sprintf("failed to push %s image to: %s", lang, containerRegistry))
+		}
+
+		pushScanner := bufio.NewScanner(pushStdout)
+		pushScanner.Split(bufio.ScanLines)
+		for pushScanner.Scan() {
+			j := pushScanner.Text()
+			fmt.Println(j)
+		}
+		if err := pushDockerImage.Wait(); err != nil {
+			log.Fatal(fmt.Sprintf("exit during pushing %s image, error: %s", lang, err))
+		}
+
+		log.Println(fmt.Sprintf("successfully pushed %s worker to %s", lang, containerRegistry))
+	}
+
+	if err := file.Close(); err != nil {
+		log.Fatalf("failed to close file")
+	}
+}
+
+// go run prepare_for_prebuilt_workers.go -l 1 go
```

Comment:
nit: what is `-l 1 go`?

Reply:
I originally wanted to take everything after -l 1 together as an argument list. It should be something like `-l 1 go python ruby` when providing the flags. I then found out that it depends on location, for this command to work, I could only have `-l 1 go python ruby` session at the end, otherwise it will take anything after this session also into the list. I updated to take multiple language as:
```
go run prepare_for_prebuilt_workers.go -l go -l python -l ruby
```

Thank you for pointing this out!